File: apps/Zivohealth/Sources/Models/Appointment.swift
import Foundation

struct Appointment: Identifiable, Codable {
    let id: Int
    let patientId: Int
    let doctorId: Int
    let consultationRequestId: Int?
    let title: String
    let description: String?
    let appointmentDate: Date
    let durationMinutes: Int
    let status: String
    let appointmentType: String
    let patientNotes: String?
    let doctorNotes: String?
    let createdAt: Date
    let updatedAt: Date
    
    enum CodingKeys: String, CodingKey {
        case id
        case patientId = "patient_id"
        case doctorId = "doctor_id"
        case consultationRequestId = "consultation_request_id"
        case title, description
        case appointmentDate = "appointment_date"
        case durationMinutes = "duration_minutes"
        case status
        case appointmentType = "appointment_type"
        case patientNotes = "patient_notes"
        case doctorNotes = "doctor_notes"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

struct AppointmentWithDetails: Identifiable, Codable {
    let id: Int
    let patientId: Int
    let doctorId: Int
    let consultationRequestId: Int?
    let title: String
    let description: String?
    let appointmentDate: Date
    let durationMinutes: Int
    let status: String
    let appointmentType: String
    let patientNotes: String?
    let doctorNotes: String?
    let createdAt: Date
    let updatedAt: Date
    let patientName: String
    let patientEmail: String
    let doctorName: String
    let doctorEmail: String
    
    enum CodingKeys: String, CodingKey {
        case id
        case patientId = "patient_id"
        case doctorId = "doctor_id"
        case consultationRequestId = "consultation_request_id"
        case title, description
        case appointmentDate = "appointment_date"
        case durationMinutes = "duration_minutes"
        case status
        case appointmentType = "appointment_type"
        case patientNotes = "patient_notes"
        case doctorNotes = "doctor_notes"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case patientName = "patient_name"
        case patientEmail = "patient_email"
        case doctorName = "doctor_name"
        case doctorEmail = "doctor_email"
    }
}

struct AppointmentCreate: Codable {
    let doctorId: Int
    let consultationRequestId: Int?
    let title: String
    let description: String?
    let appointmentDate: Date
    let durationMinutes: Int
    let status: String
    let appointmentType: String
    let patientNotes: String?
    
    enum CodingKeys: String, CodingKey {
        case doctorId = "doctor_id"
        case consultationRequestId = "consultation_request_id"
        case title, description
        case appointmentDate = "appointment_date"
        case durationMinutes = "duration_minutes"
        case status
        case appointmentType = "appointment_type"
        case patientNotes = "patient_notes"
    }
}

struct AppointmentUpdate: Codable {
    let title: String?
    let description: String?
    let appointmentDate: Date?
    let durationMinutes: Int?
    let status: String?
    let appointmentType: String?
    let patientNotes: String?
    let doctorNotes: String?
    
    enum CodingKeys: String, CodingKey {
        case title, description
        case appointmentDate = "appointment_date"
        case durationMinutes = "duration_minutes"
        case status
        case appointmentType = "appointment_type"
        case patientNotes = "patient_notes"
        case doctorNotes = "doctor_notes"
    }
} 

File: backend/app/schemas/appointment.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel

# Shared properties
class AppointmentBase(BaseModel):
    title: str
    description: Optional[str] = None
    appointment_date: datetime
    duration_minutes: int = 30
    status: str = "scheduled"
    appointment_type: str = "consultation"
    patient_notes: Optional[str] = None
    doctor_notes: Optional[str] = None

# Properties to receive on appointment creation
class AppointmentCreate(AppointmentBase):
    doctor_id: int
    consultation_request_id: Optional[int] = None

# Properties to receive on appointment update
class AppointmentUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    appointment_date: Optional[datetime] = None
    duration_minutes: Optional[int] = None
    status: Optional[str] = None
    appointment_type: Optional[str] = None
    patient_notes: Optional[str] = None
    doctor_notes: Optional[str] = None

# Properties shared by models stored in DB
class AppointmentInDBBase(AppointmentBase):
    id: int
    patient_id: int
    doctor_id: int
    consultation_request_id: Optional[int] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

# Properties to return to client
class Appointment(AppointmentInDBBase):
    pass

# Properties stored in DB
class AppointmentInDB(AppointmentInDBBase):
    pass

# Appointment with user details
class AppointmentWithDetails(Appointment):
    patient_name: str
    patient_email: str
    doctor_name: str
    doctor_email: str 